### PR TITLE
refactor: private cluster host agent without go templates

### PR DESCRIPTION
**Pull request by alexeldeib**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/agent/testdata/AKSUbuntu1604+EnablePrivateClusterHostsConfigAgent/CSECommand [hunk 1]
```diff
@@ -1 +1 @@
...
...d64-v1.1.3.tgz CNI_PLUGINS_URL=https://acs-mirror.azureedge.net/cni/cni-plugins-amd64-v0.7.6.tgz CLOUDPROVIDER_BACKOFF=<nil> CLOUDPROVIDER_BACKOFF_MODE= CLOUDPROVIDER_BACKOFF_RETRIES=0 CLOUDPROVIDER_BACKOFF_EXPONENT=0 CLOUDPROVIDER_BACKOFF_DURATION=0 CLOUDPROVIDER_BACKOFF_JITTER=0 CLOUDPROVIDER_RATELIMIT=<nil> CLOUDPROVIDER_RATELIMIT_QPS=0 CLOUDPROVIDER_RATELIMIT_QPS_WRITE=0 CLOUDPROVIDER_RATELIMIT_BUCKET=0 CLOUDPROVIDER_RATELIMIT_BUCKET_WRITE=0 LOAD_BALANCER_DISABLE_OUTBOUND_SNAT=<nil> USE_MANAGED_IDENTITY_EXTENSION=false USE_INSTANCE_METADATA=false LOAD_BALANCER_SKU= EXCLUDE_MASTER_FROM_STANDARD_LB=true MAXIMUM_LOADBALANCER_RULE_COUNT=0 CONTAINER_RUNTIME= CLI_TOOL= CONTAINERD_DOWNLOAD_URL_BASE=https://storage.googleapis.com/cri-containerd-release/ NETWORK_MODE= KUBE_BINARY_URL= USER_ASSIGNED_IDENTITY_ID=userAssignedID API_SERVER_NAME= IS_VHD=true GPU_NODE=false SGX_NODE=false MIG_NODE=false CONFIG_GPU_DRIVER_IF_NEEDED=true ENABLE_GPU_DEVICE_PLUGIN_IF_NEEDED=false TELEPORTD_PLUGIN_DOWNLOAD_URL= CONTAINERD_VERSION= CONTAINERD_PACKAGE_URL= RUNC_VERSION= RUNC_PACKAGE_URL= ENABLE_HOSTS_CONFIG_AGENT="true" /usr/bin/nohup /bin/bash -c "/bin/bash /opt/azure/containers/provision_start.sh"
```
Comment (alexeldeib) [inline pkg/agent/testdata/AKSUbuntu1604+EnablePrivateClusterHostsConfigAgent/CSECommand:1]: n.b. `ENABLE_HOSTS_CONFIG_AGENT="true"` towards the very end right before the nohup

**Comment by alexeldeib** [comment]
ado issues, seems like arm was fine but rerunning for sanity

File: parts/linux/cloud-init/artifacts/cse_main.sh [hunk 2]
```diff
@@ -215,11 +215,11 @@ if [[ "{{GetTargetEnvironment}}" == "AzureChinaCloud" ]]; then
...
 
-{{- if EnableHostsConfigAgent}}
-logs_to_events "AKS.CSE.configPrivateClusterHosts" configPrivateClusterHosts
-{{- end}}
+if [[ "${ENABLE_HOSTS_CONFIG_AGENT}" == "true" ]]; then
+    logs_to_events "AKS.CSE.configPrivateClusterHosts" configPrivateClusterHosts
+fi
 
-{{- if ShouldConfigTransparentHugePage}}
+{{ if ShouldConfigTransparentHugePage -}}
```
Comment (cameronmeissner) [inline parts/linux/cloud-init/artifacts/cse_main.sh:222]: what ends up being the diff here?

**Review by cameronmeissner — APPROVED** [review]
overall lgtm, makes sense to slowly be moving away from go templates

File: parts/linux/cloud-init/artifacts/cse_main.sh (hunk 2 above)
Comment (alexeldeib) [reply parts/linux/cloud-init/artifacts/cse_main.sh:222]: whitespace in generated files

the minus before the if leads to a missing newline between some bash
the minus after the if avoids removing the preceding newline, but trips everything inside the brackets

**Pull request merged**